contracts/JBMultiTerminal.sol

```diff
@@ -549,6 +547,161 @@ contract JBMultiTerminal is JBOperatable, Ownable, ERC2771Context, IJBMultiTermi
         }
     }
 
+    /// @notice Process a fee of the specified amount from a project.
+    /// @dev Only accepts calls from this terminal itself.
+    /// @param _projectId The project ID the fee is being paid from.
+    /// @param _token The token the fee is being paid in.
+    /// @param _amount The fee amount, as a floating point number with 18 decimals.
+    /// @param _beneficiary The address to mint the platform's tokens for.
+    /// @param _feeTerminal The terminal that'll receive the fees. This'll be filled if one isn't provided.
+    function executeProcessFee(
+        uint256 _projectId,
+        address _token,
+        uint256 _amount,
+        address _beneficiary,
+        IJBPaymentTerminal _feeTerminal
+    ) external {
```

Comment:
why external not internal? (to make it catchable, I assume?) 
 unfortunate tradeoff lol